### ISSUE TITLE
First page compilation failed

### ISSUE THREAD

**Issue by willvieira** [PoisotLab/PLMT#13]
Hi Tim,

I’ve tried to use your template, it compiles fine until I change something either in the `infos.yaml`, `authors.yaml` or `ABSTRACT` files.
I saw it compiled all the information into the `.metadata.yaml` without problems, however, these infos does not come in the first page of the outputs (.pdf, .odf), but in the end of it after the References section (I attached an example where I just changed the title of the manuscript).

Every time I change something in one of these files (infos, authors or abstract), I have the following error:

```makefile
[1] "manuscript.md"  	
node .plmt/metadata.js  
sed -i '1s/^/---\n/' .metadata.yaml  
sed: 1: ".metadata.yaml": invalid command code .     
make: *** [.metadata.yaml] Error 1  
```

And then the title, authors and abstract do not come out.

I’m running on:

```makefile
OS X 10.12.5
pdfTeX 3.14159265-2.6-1.40.17 (TeX Live 2016)
pandoc 1.19.2.1
```

Thanks!
[manuscript_preprint.pdf](https://github.com/PoisotLab/PLMT/files/1131136/manuscript_preprint.pdf)

**Comment by tpoisot** [comment]
This seems to be a difference between sed on OS X and sed on Linux. I'll have a look at the documentation for both, and see how I can update the Makefile. 

Thanks for reporting.

**Comment by tpoisot** [comment]
Possible solution:

https://stackoverflow.com/a/38595160

TODO: setup Travis with Linux + OS X?

**Comment by tpoisot** [comment]
Can you download the new version and let me know?
https://github.com/PoisotLab/PLMT/archive/v0.2.0.zip

**Comment by willvieira** [comment]
Yes, I've done the same yesterday and is working fine on OSX with this change.

However, the line break `\n` on `sed -i "" -e '1s/^/---\n/' $@` is not working, which results in this in the `.metadata.yaml` file: 
```yaml
---ntitle: Template for pandoc / markdown manuscripts
```
instead of:
```yaml
---
title: Template for pandoc / markdown manuscripts
```

I've tried different ways based in this answer and others: 
https://stackoverflow.com/questions/3790454/in-yaml-how-do-i-break-a-string-over-multiple-lines
but I still not able to make it work.

Do you have any suggestion? 

Thanks again!

**Comment by tpoisot** [comment]
Can you try with 

~~~
sed -i "" -e "1s/^/---\n/"
~~~

and let me know?

**Comment by willvieira** [comment]
Doesn't work either.

**Comment by KevCaz** [comment]
I remember having some hard times because of the differences between the two sed. Now when I am on macOS I often use the gnu-sed. 

If you are a brew user => `brew install gnu-sed` and use `gsed` instead of `sed` (or create an alias).

**Comment by willvieira** [comment]
Thanks for the help @KevCaz!
Using `gsed` is working fine now.

The question if it still portable to both OSX and Linux.

**Comment by KevCaz** [comment]
- Using gsed = 1 more requirement on MacOS and use the almost the same expression (gsed instead of sed)
- Using the  *native sed* on MacOS = you have to tune the expression (a few tweaks I guess)

**Comment by tpoisot** [comment]
I think in both cases, there can be a global variable that will be either `sed` or `gsed` depending if the `uname` has `Darwin` in it (OS X) or not. I don't have a mac nearby to test it.

**Comment by KevCaz** [comment]
As expected: 

https://stackoverflow.com/questions/9279765/how-to-detect-os-in-a-make-file

**Comment by KevCaz** [comment]
So, I guess this should work ... or something similar... 

```
OS := $(shell uname)
ifeq ($(OS),Darwin)
  sed = gsed 
else
  sed = sed 
endif
```

then use $(sed)
